modules/sfm/CMakeLists.txt

```diff
@@ -0,0 +1,83 @@
+set(the_description "SFM algorithms")
+
+ocv_warnings_disable(CMAKE_CXX_FLAGS -Wundef -Winvalid-pch)
+
+# needed since we are linking static libs
+set(CMAKE_CXX_FLAGS ${CMAKE_CSS_FLAGS} "-fPIC")
+
+
+### LIBMV LIGHT DEFINITIONS ###
+
+set(LIBMV_LIGHT_INCLUDES
+  src/libmv_light
+  src/libmv_light/third_party/ssba
+)
+
+set(LIBMV_LIGHT_LIBS
+  reconstruction
+  correspondence
+  camera
+  multiview
+  multiview_test_data
+  numeric
+)
+
+
+### LIBMV LIGHT EXTERNAL DEPENDENCIES ###
+
+find_package( Ceres )
+
+if ( Ceres_FOUND )
+  add_definitions("-DCERES_FOUND=1")
+  list(APPEND LIBMV_LIGHT_LIBS simple_pipeline)
+else ()
+  add_definitions("-DCERES_FOUND=0")
+endif ()
+
+
+### CHECK OPENCV SFM MODULE DEPENDENCIES ###
+
+set(OPENCV_SFM_DEPS
+  opencv_core
+  opencv_ts
+  opencv_calib3d
+  opencv_features2d
+  opencv_xfeatures2d
+  opencv_viz
+)
+
+ocv_check_dependencies(${OPENCV_SFM_DEPS})
+ocv_include_directories("${OpenCV_SOURCE_DIR}/include/opencv")
+ocv_include_modules_recurse(${OPENCV_SFM_DEPS})
```

Comment:
This will not work. It is not a sample. There are some special rules for modules (required for proper dependency handling). Refer to structure in this this:
https://github.com/Itseez/opencv/blob/09b9b0fb9e9c9dd8c9e0d65705f8f19aa4c27f8a/modules/viz/CMakeLists.txt
Especially on `ocv_module_disable()`.

P.S. I see right lines of below, so try to remove these lines.